### PR TITLE
CLN: Continued indexes/test_base.py cleanup

### PR DESCRIPTION
**Pull request by WillAyd**
progress towards #20812

### CODE REVIEW

File: pandas/tests/indexes/test_base.py [hunk 1]
```diff
@@ -678,61 +678,53 @@ def test_intersection(self):
...
-        expected = Index(['A', 'B', 'A'], dtype='object')
-        result = idx1.intersection(idx2)
-        tm.assert_index_equal(result, expected)
+        assert res.name == exp.name
+        tm.assert_index_equal(res, exp)
 
-        # preserve names
+    @pytest.mark.parametrize("first_nm,sec_nm,exp_nm", [
+        ('A', 'A', 'A'), ('A', 'B', None), (None, 'B', None)])
+    def test_intersection_name_preservation2(self, first_nm, sec_nm, exp_nm):
```
Comment (WillAyd) [inline pandas/tests/indexes/test_base.py]: Kept as a separate test for now, but ideally should be combined with the test above as they cover the same topics. Waiting until I get around to replacing the index class attrs with fixtures to explore in more detail

File: pandas/tests/indexes/test_base.py [hunk 2]
```diff
@@ -878,24 +845,23 @@ def test_map_with_tuples(self):
...
         #   returns an Index.
         first_level = ['foo', 'bar', 'baz']
         multi_index = MultiIndex.from_tuples(lzip(first_level, [1, 2, 3]))
         reduced_index = multi_index.map(lambda x: x[0])
         tm.assert_index_equal(reduced_index, Index(first_level))
 
-    def test_map_tseries_indices_return_index(self):
-        date_index = tm.makeDateIndex(10)
+    @pytest.mark.parametrize("attr", [
+        'makeDateIndex', 'makePeriodIndex', 'makeTimedeltaIndex'])
```
Comment (WillAyd) [inline pandas/tests/indexes/test_base.py]: Will most likely replace with a fixture in the future, but made sense to parametrize for now

File: pandas/tests/indexes/test_base.py [hunk 3]
```diff
@@ -996,30 +958,38 @@ def test_iadd_string(self):
         index += '_x'
         assert 'a_x' in index
 
-    def test_difference(self):
-
+    @pytest.mark.parametrize("second_nm,exp", [
+        (None, None), ('name', 'name')])
```
Comment (WillAyd) [inline pandas/tests/indexes/test_base.py]: Might be over-engineering the parameters in this case, but figure if we wanted to expand to cover more scenarios this is a better framework to do so

File: pandas/tests/indexes/test_base.py [hunk 4]
```diff
@@ -1092,25 +1058,23 @@ def test_intersection_difference(self):
...
-        assert not self.floatIndex.is_object()
-
-    def test_is_all_dates(self):
-        assert self.dateIndex.is_all_dates
-        assert not self.strIndex.is_all_dates
-        assert not self.intIndex.is_all_dates
+    @pytest.mark.parametrize("attr,exp", [
+        ('strIndex', False), ('boolIndex', False), ('catIndex', False),
+        ('intIndex', True), ('dateIndex', False), ('floatIndex', True)])
+    def test_is_numeric(self, attr, exp):
```
Comment (WillAyd) [inline pandas/tests/indexes/test_base.py]: This and the two tests below it did not consistently test all of the index types so I've added where missing for better coverage. Should ultimately be replaced with fixtures but same concept will apply

File: pandas/tests/indexes/test_base.py [hunk 5]
```diff
@@ -1142,19 +1106,21 @@ def test_format(self):
             expected = [str(index[0])]
             assert formatted == expected
 
+        self.strIndex[:0].format()
+
+    def test_format_missing(self, nulls_fixture):
```
Comment (WillAyd) [inline pandas/tests/indexes/test_base.py]: Was only testing `np.nan` and `None` before but figured this was a good use case for the new nulls fixture

File: pandas/tests/indexes/test_base.py [hunk 6]
```diff
@@ -1142,19 +1106,21 @@ def test_format(self):
...
         formatted = index.format()
         expected = [str(index[0]), str(index[1]), u('NaN')]
         assert formatted == expected
 
-        # is this really allowed?
-        index = Index([1, 2.0 + 3.0j, None])
-        formatted = index.format()
-        expected = [str(index[0]), str(index[1]), u('NaN')]
-        assert formatted == expected
+    def test_format_missing_obj(self, nulls_fixture):
```
Comment (WillAyd) [inline pandas/tests/indexes/test_base.py]: Trying to balance too many changes but am leaning towards combining this with the above test. Both are pretty similar but leave strange gaps in what they test across the float and object types

File: pandas/tests/indexes/test_base.py [hunk 7]
```diff
@@ -678,61 +678,53 @@ def test_intersection(self):
...
 
-        idx2 = Index(['B', 'D', 'A'])
-        expected = Index(['A', 'B', 'A'], dtype='object')
-        result = idx1.intersection(idx2)
-        tm.assert_index_equal(result, expected)
+        assert res.name == exp.name
+        tm.assert_index_equal(res, exp)
 
-        # preserve names
+    @pytest.mark.parametrize("first_nm,sec_nm,exp_nm", [
```
Comment (jreback) [inline pandas/tests/indexes/test_base.py]: ideally can you spell things out a bit more, e.g. ``first_name, second_name, expected_name`` are just fine

File: pandas/tests/indexes/test_base.py [hunk 8]
```diff
@@ -678,61 +678,53 @@ def test_intersection(self):
...
+        first.name = first_nm
+        second.name = sec_nm
         intersect = first.intersection(second)
-        assert intersect.name == 'A'
+        assert intersect.name == exp_nm
 
-        second.name = 'B'
-        intersect = first.intersection(second)
-        assert intersect.name is None
+    @pytest.mark.parametrize("idx2,keeps_nm", [
```
Comment (jreback) [inline pandas/tests/indexes/test_base.py]: so ``keeps_nm`` -> ``keeps_name`` and normalize the input params, e.g. ``index`` (or index2 or other) as appropriate

File: pandas/tests/indexes/test_base.py [hunk 9]
```diff
@@ -744,19 +736,31 @@ def test_intersect_str_dates(self):
...
     def test_union(self):
+        # TODO: Replace with fixtures
         first = self.strIndex[5:20]
         second = self.strIndex[:10]
         everything = self.strIndex[:20]
+
         union = first.union(second)
         assert tm.equalContents(union, everything)
 
+    @pytest.mark.parametrize("klass", [
```
Comment (jreback) [inline pandas/tests/indexes/test_base.py]: this notion of klass we have all over the place, sometimes its also called box (not necessary to do here, just letting you know, prob best to update the issue you created)

File: pandas/tests/indexes/test_base.py [hunk 10]
```diff
@@ -766,61 +770,23 @@ def test_union(self):
...
-
-        first = Index(list('ab'), name='A')
-        second = Index([])
-        union = first.union(second)
-        expected = Index(list('ab'), name='A')
-        tm.assert_index_equal(union, expected)
-
-        first = Index([], name='A')
-        second = Index(list('ab'))
+    @pytest.mark.parametrize("first,second,nm", [
```
Comment (jreback) [inline pandas/tests/indexes/test_base.py]: maybe could replace this with 2 parameterize (for a cartesian product of the inputs)

File: pandas/tests/indexes/test_base.py [hunk 11]
```diff
@@ -967,18 +933,14 @@ def test_append_multiple(self):
         result = index.append([])
         tm.assert_index_equal(result, index)
 
-    def test_append_empty_preserve_name(self):
+    @pytest.mark.parametrize("nm,exp", [
```
Comment (jreback) [inline pandas/tests/indexes/test_base.py]: ``name, expected``

File: pandas/tests/indexes/test_base.py (hunk 3 above)
Comment (jreback) [reply pandas/tests/indexes/test_base.py]: ok by me

File: pandas/tests/indexes/test_base.py (hunk 6 above)
Comment (jreback) [reply pandas/tests/indexes/test_base.py]: sure

File: pandas/tests/indexes/test_base.py [hunk 12]
```diff
@@ -678,61 +678,53 @@ def test_intersection(self):
         inter = first.intersection(first)
         assert inter is first
 
+    @pytest.mark.parametrize("idx2,keeps_nm", [
+        (Index([3, 4, 5, 6, 7], name="idx"), True),  # preserve same name
+        (Index([3, 4, 5, 6, 7], name="other"), False),  # drop diff names
+        (Index([3, 4, 5, 6, 7]), False)])
+    def test_intersection_name_preservation(self, idx2, keeps_nm):
```
Comment (jreback) [inline pandas/tests/indexes/test_base.py]: side note, we prob need to split out test_base into multiple files its growing pretty big (do this as a separate step), these might be things like: test_indexing.py, test_missing.py, test_set_ops.py etc, in essence mirroring the various sub-files we do for series ops

**Review by jreback — CHANGES_REQUESTED** [review]
looks good. if can fix the naming then looks good to merge

File: pandas/tests/indexes/test_base.py (hunk 12 above)
Comment (WillAyd) [reply pandas/tests/indexes/test_base.py]: Yep - I have this as the last TODO in the referenced issue so we don't lose sight of it

File: pandas/tests/indexes/test_base.py (hunk 7 above)
Comment (WillAyd) [reply pandas/tests/indexes/test_base.py]: Would you prefer generally using `result` and `expected` over `res` and `exp`? I've seen both and typically use the latter but if we have a preference we could start being explicit about those standards (maybe in TESTING_README)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @WillAyd

**Pull request merged**